gems/laa_multi_step_forms/docs/DSLDecisionTree.md

```diff
@@ -0,0 +1,136 @@
+# DSL Desicion Tree
+
+## What?
+
+This is a simple DSL that layers on top of the existing DecisonTree code that was used in crime apply. Due to
+this it is fully backwards compatable with the previous implementation.
+
+## Why?
+
+This was created to simplify the expression of the logical flow of the system. Basically make it easiler to
+read, understand and update the flow compared to the previous implementation. This has been coupled with an update to
+the testing helper to again try and make the tests easy to read, understand and update.
+
+## How does it work?
+
+The basic concept of the descison tree is that given the user has come from some location `A` they are redirected
+to a new location `B`, this can then be complicated by adding logic such that they would instead be redirected to
+a different location `C` if a specificed condition is `X` met.
+
+Hense the basic syntax of the DSl is for the previous example is:
+
+```
+from(A)
+  .when(X).goto(show: C)
+  .goto(show: B)
+```
+
+The below could be written on a single line, however breaking it done in `when` and `goto` pairs makes it easier
+to parse, the logic will return the result of the first `goto` after a `when` condition has been met - notice the
+final `goto` does not require a `when` condition as it catches all remaining choices.
+
+## Downsides
+
+The main downside of this approach is that the DSL is loaded within the Class object - this means we only need to
+create a single instance of it on boot, but that dos result in slower boot times which may be a concern as the
+ruleset grows in the future - or multiple large rulesets exist.
```

Comment:
```suggestion
create a single instance of it on boot, but that does result in slower boot times. This may become a concern as the
ruleset grows in the future - or multiple large rulesets exist.
```